tensorizer/stream_io.py

```diff
@@ -430,15 +461,15 @@ def _read_until(
             if ba is None:
                 rq_sz = goal_position - self._curr
                 if self._end is not None and self._curr + rq_sz > self._end:
-                    rq_sz = self._end - self._curr
+                    rq_sz = self._end - self._curr + 1
```

Comment:
Hmm, did we have an off by one here?

Reply:
Indeed; I mentioned it in [a comment on #87](https://github.com/coreweave/tensorizer/pull/87#pullrequestreview-1908134052):

> HTTP `Range` requests have inclusive endpoints—e.g., a range of 0–1 is two bytes, not one. None of our code had been handling this correctly, but the multiple ways it got it wrong simultaneously had perfectly aligned to make it appear to behave correctly anyway. In #99, I changed it to handle end markers correctly, but I wonder if we should just formally change the `open_stream` and `CURLStreamFile` interfaces instead, so that their end parameters are exclusive of the final byte, like other slicing behaves in Python.